File: python/src/ev1/ut2_controlflujometodos/u01condicionales/teoria/U04_CondicionalesAnidados.py
"""
U04 — Condicionales anidados y operadores logicos.

OBJETIVO:
  - if dentro de otro if (anidado) cuando hay dos condiciones en cascada.
  - and: ambas deben ser True. or: al menos una. not: invierte.
  - Combinar con if/elif cuando match no encaja (rangos, desigualdades).

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
"""
print("\n¡INICIO DEL PROGRAMA!\n")
"""
PRIMERA PARTE - Ejemplo con "if" dentro de otros "if".
"""
print(f"\n¡PRUEBA USANDO UN \"IF\" ANIDADO DENTRO DE OTRO \"IF\"!\n")
edad = 20
tiene_carnet = True

# PRIMERA VALIDACIÓN: Si ya es mayor de edad el if se ejecuta.
if edad >= 18:
    # SEGUNDA VALIDACIÓN: Si además de ser mayor, tiene carnet, el if se ejecuta.
    if tiene_carnet:
        print("Puede conducir")
    else:
        print("Necesita sacarse el carnet")
else:
    print("Es menor de edad")

"""
SEGUNDA PARTE - Solo si ambas condiciones se cumplen usando "and" se ejeucta el "if".
"""
print(f"\n¡PRUEBA USANDO UN \"IF\" CON MULTIPLES CONDICIONES ENCADENADAS CON \"AND\"!\n")
temperatura = 25
soleado = True
if temperatura > 20 and soleado:
    print("Buen dia para salir")
"""
TERCERA PARTE - Si una (cualquiera) de las dos condiciones se cumple usando "or" se ejecuta el "if".
"""
print(f"\n¡PRUEBA USANDO UN \"IF\" CON MULTIPLES CONDICIONES ENCADENADAS CON \"OR\"!\n")
es_finde = True
es_festivo = False

if es_finde or es_festivo:
    print("Dia de descanso")

"""
CUARTA PARTE - Si una condición no se cumple (validando con not), entonces el "if" se ejecuta.
"""
print(f"\n¡PRUEBA USANDO UN \"IF\" NEGANDO CONDICIONES CON \"NOT\"!\n")
llueve = False
if not llueve:
    print("No hace falta paraguas")

print("\n¡FIN DEL PROGRAMA!\n")

File: cpp/src/ev1/ut2_controlflujometodos/u02condicionales/teoria/U04_CondicionalesAnidados.cpp
/*
U04 — Condicionales anidados: if dentro de if.

OBJETIVO:
  - Combinar varias condiciones en niveles (anidamiento).
  - Decidir primero una condicion general y luego otra mas especifica.
  - Usar llaves { } para claridad cuando hay varios niveles.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

int main() {
    cout << "\n¡INICIO DEL PROGRAMA!\n" << endl;

    /*
    PRIMERA PARTE - Anidado simple: aprobado y luego sobresaliente.
    */
    cout << "¡DEMO 1: NOTA CON DOS NIVELES!\n" << endl;

    int nota = 9;
    if (nota >= 5)
    {
        if (nota >= 9)
        {
            cout << "Nota " << nota << ": Sobresaliente" << endl;
        }
        else
        {
            cout << "Nota " << nota << ": Aprobado (sin sobresaliente)" << endl;
        }
    }
    else
    {
        cout << "Nota " << nota << ": Suspenso" << endl;
    }

    nota = 6;
    if (nota >= 5)
    {
        if (nota >= 9)
        {
            cout << "Sobresaliente" << endl;
        }
        else
        {
            cout << "Nota " << nota << ": Aprobado" << endl;
        }
    }
    else
    {
        cout << "Suspenso\n" << endl;
    }

    /*
    SEGUNDA PARTE - Edad y permiso de conducir con licencia.
    */
    cout << "¡DEMO 2: EDAD Y LICENCIA!\n" << endl;

    int edad = 20;
    bool tieneLicencia = true;

    if (edad >= 18)
    {
        if (tieneLicencia)
        {
            cout << "Puede conducir legalmente." << endl;
        }
        else
        {
            cout << "Es mayor de edad pero necesita sacarse la licencia." << endl;
        }
    }
    else
    {
        cout << "Es menor de edad: no puede conducir." << endl;
    }

    edad = 16;
    tieneLicencia = false;
    if (edad >= 18)
    {
        if (tieneLicencia)
        {
            cout << "Puede conducir." << endl;
        }
        else
        {
            cout << "Necesita licencia." << endl;
        }
    }
    else
    {
        cout << "Con " << edad << " años aun no puede conducir.\n" << endl;
    }

    /*
    TERCERA PARTE - Acceso a sistema: usuario valido y rol admin.
    */
    cout << "¡DEMO 3: USUARIO Y ROL!\n" << endl;

    bool usuarioValido = true;
    bool esAdmin = false;

    if (usuarioValido)
    {
        cout << "Sesion iniciada correctamente." << endl;
        if (esAdmin)
        {
            cout << "Panel de administracion desbloqueado." << endl;
        }
        else
        {
            cout << "Acceso de usuario estandar." << endl;
        }
    }
    else
    {
        cout << "Error: credenciales incorrectas." << endl;
    }

    cout << "\n¡FIN DEL PROGRAMA!\n" << endl;
    return 0;
}
